src/Repository/LineItemRepository.php

```diff
@@ -153,4 +156,31 @@ private function applyCriterias(FindLineItemCriteria $criteria, QueryBuilder $qu
                 ->setParameter('endAt', $criteria->getLineItemEndAt());
         }
     }
+
+    public function createOrUpdate(LineItem $lineItem): LineItem
+    {
+        try {
+            $this->persist($lineItem);
+            $this->flush();
+
+            return $lineItem;
+        } catch (UniqueConstraintViolationException $exception) {
+            $this->registry->resetManager();
+
+            /** @var LineItem $stored */
+            $stored = $this->findOneBy(['slug' => $lineItem->getSlug()]);
+
+            $stored->setUri($lineItem->getUri());
+            $stored->setLabel($lineItem->getLabel() ? : '');
+            $stored->setIsActive($lineItem->isActive());
+            $stored->setMaxAttempts($lineItem->getMaxAttempts());
+            $stored->setStartAt($lineItem->getStartAt());
+            $stored->setEndAt($lineItem->getEndAt());
+
+            $this->persist($stored);
+            $this->flush();
+
+            return $stored;
+        }
+    }
```

Comment:
IMO: update part is not needed there, at least for the current iteration.
Reason: publishing side ATM doesn't send all this data, so there's no sense in updating anything. I would leave it as (pseudo-code):
```
try {
    to create;
} catch (\AlreadyExistingLineItem $e) {
    return $alreadyExisting;
}